src/client/arg_parse.c

```diff
@@ -36,68 +32,85 @@ int _main(int argc, char **argv) {
         switch (c) {
 
         case 'a':
-            arg_a = 1;
-            strcpy(path, optarg);
+            if (ret->cmd != -1){
+                return -1;
+            }
+            ret->cmd = CMD_ADD;
+            ret->path = malloc(strlen(optarg) + 1);
+            strcpy(ret->path, optarg);
+            ret->priority = (int8_t)2;
             break;
 
         case 'p':
-            arg_p = 1;
-            strcpy(p_value, optarg);
+            if (ret->cmd == CMD_ADD){
+                ret->priority = atoi(optarg);
+            }
+            else if (ret->cmd == -1){
+                ret->cmd = CMD_PRINT;
+                ret->job_id = atoi(optarg);
```

Comment:
`job_id` is `int64_t`, so `atoll` should be used. But `strtoll` would be better for error handling.

Same for all other instances of `atoi`.